blobfs/src/sync_io.rs

```diff
@@ -60,31 +67,32 @@ impl BlobFs {
 
         trace!("load_chunks_on_demand: blob_id {:?}", blob_id);
 
-        Ok((blob_id.to_os_string().into_string().unwrap(), st.st_size))
+        Self::check_st_size(&blob_id_full_path, st.st_size)?;
+
+        Ok((
+            blob_id.to_os_string().into_string().unwrap(),
+            st.st_size as u64,
+        ))
     }
 
     #[cfg(feature = "virtiofs")]
-    fn load_chunks_on_demand(&self, inode: Inode, foffset: u64) -> io::Result<()> {
+    fn load_chunks_on_demand(&self, inode: Inode, offset: u64) -> io::Result<()> {
         // prepare BlobPrefetchRequest and call device.prefetch().
         // Make sure prefetch doesn't use delay_persist as we need the
         // data immediately.
         let (blob_id, size) = self.get_blob_id_and_size(inode)?;
-        let offset: u32 = foffset.try_into().map_err(|_| {
-            einval!(format!(
-                "blobfs: load_chunks_on_demand: foffset {} is larger than u32::MAX",
-                foffset
-            ))
-        })?;
-        let len = (size - offset as i64).try_into().map_err(|_| {
-            einval!(format!(
-                "blobfs: load_chunks_on_demand: len {} is larger than u32::MAX",
-                (size - offset as i64)
-            ))
-        })?;
+        if size < offset {
```

Comment:
what about `if size == offset`?

Reply:
the BlobPrefetchRequest.len will be 0,  storage mod should handle it correctly, let me check to make sure.